pkg/apiserver/webhook/cloudentityselector_webhook.go

```diff
@@ -98,142 +106,196 @@ func (r *CloudEntitySelector) Default() {
-			return
+			v.Log.Error(err, "failed to set owner account", "cloudentityselector", selector, "account", *accountNameSpacedName)
+			return admission.Errored(http.StatusBadRequest, err)
 		}
 	}
+
+	marshaledSelector, err := json.Marshal(selector)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledSelector)
+}
+
+// InjectDecoder injects the decoder.
+func (v *CESMutator) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
 }
 
 // TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
 // nolint:lll
 // +kubebuilder:webhook:verbs=create;update,path=/validate-crd-cloud-antrea-io-v1alpha1-cloudentityselector,mutating=false,failurePolicy=fail,groups=crd.cloud.antrea.io,resources=cloudentityselectors,versions=v1alpha1,name=vcloudentityselector.kb.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 
-var _ webhook.Validator = &CloudEntitySelector{}
+// CESValidator is used to validate CES object.
+type CESValidator struct {
+	Client  controllerclient.Client
+	Log     logr.Logger
+	decoder *admission.Decoder
+}
+
+// Handle handles validator admission requests for CES.
+func (v *CESValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	v.Log.V(1).Info("Received CES admission webhook request", "Name", req.Name, "Operation", req.Operation)
+	switch req.Operation {
+	case admissionv1.Create:
+		return v.validateCreate(req)
+	case admissionv1.Update:
+		return v.validateUpdate(req)
+	case admissionv1.Delete:
+		return v.validateDelete(req)
+	default:
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf(errorMsgInvalidRequest))
+	}
+}
 
-// ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
-func (r *CloudEntitySelector) ValidateCreate() error {
-	cloudentityselectorlog.Info("validate create", "name", r.Name)
+// InjectDecoder injects the decoder.
+func (v *CESValidator) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
 
+// ValidateCreate implements webhook validations for CES add operation.
+func (v *CESValidator) validateCreate(req admission.Request) admission.Response {
+	selector := &v1alpha1.CloudEntitySelector{}
+	err := v.decoder.Decode(req, selector)
+	if err != nil {
+		v.Log.Error(err, "Failed to decode CloudEntitySelector", "CESValidator", req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
 	// make sure owner exists. Default will try to populate if owner with provided account name exists.
-	if r.GetObjectMeta().GetOwnerReferences() == nil {
-		return fmt.Errorf("owner account %v not found", r.Spec.AccountName)
+	if selector.GetObjectMeta().GetOwnerReferences() == nil {
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("owner account %v not found",
```

Comment:
Can we define a constant for the error message, "failed to find owner account" and re-use?
There are a couple of references of the similar error msg in this file.

Reply:
done